gui/src-tauri/src/main.rs

```diff
@@ -68,88 +69,128 @@ fn main() -> Result<()> {
+	check_environment_variables();
 
 	// Spawn server process
 	let exit_flag = Arc::new(AtomicBool::new(false));
 	let backend = Arc::new(Mutex::new(Option::<CommandChild>::None));
-	let backend_termination = backend.clone();
-	let run_path = get_launch_path(cli);
 
-	let server_info = if let Some(p) = run_path {
+	let server_info = execute_server(cli)?;
+	let build_result = setup_tauri(
+		tauri_context,
+		server_info,
+		exit_flag.clone(),
+		backend.clone(),
+	);
+
+	tauri_build_result(build_result, exit_flag, backend);
+
+	Ok(())
+}
+
+fn setup_logger(context: &tauri::Context) -> Result<flexi_logger::LoggerHandle> {
+	use flexi_logger::{
+		Age, Cleanup, Criterion, Duplicate, FileSpec, Logger, Naming, WriteMode,
+	};
+	use tauri::Error;
+
+	// Based on https://docs.rs/tauri/2.0.0-alpha.10/src/tauri/path/desktop.rs.html#238-256
+	#[cfg(target_os = "macos")]
+	let path = dirs_next::home_dir().ok_or(Error::UnknownPath).map(|dir| {
+		dir.join("Library/Logs")
+			.join(&context.config().identifier)
+	});
+
+	#[cfg(not(target_os = "macos"))]
+	let path = dirs_next::data_dir()
+		.ok_or(Error::UnknownPath)
+		.map(|dir| dir.join(&context.config().identifier).join("logs"));
+
+	Ok(Logger::try_with_env_or_str("info")?
+		.log_to_file(FileSpec::default().directory(path.expect("We need a log dir")))
+		.format_for_files(|w, now, record| util::logger_format(w, now, record, false))
+		.format_for_stderr(|w, now, record| util::logger_format(w, now, record, true))
+		.rotate(
+			Criterion::Age(Age::Day),
+			Naming::Timestamps,
+			Cleanup::KeepLogFiles(2),
+		)
+		.duplicate_to_stderr(Duplicate::All)
+		.write_mode(WriteMode::BufferAndFlush)
+		.start()?)
+}
+#[cfg(windows)]
+fn setup_webview2() -> Result<()> {
+	use crate::util::webview2_exists;
+	use win32job::{ExtendedLimitInfo, Job};
+
+	let mut info = ExtendedLimitInfo::new();
+	info.limit_kill_on_job_close();
+	let job = Job::create_with_limit_info(&mut info).expect("Failed to create Job");
+	job.assign_current_process()
+		.expect("Failed to assign current process to Job");
+
+	// We don't do anything with the job anymore, but we shouldn't drop it because that would
+	// terminate our process tree. So we intentionally leak it instead.
+	std::mem::forget(job);
+
+	if !webview2_exists() {
+		// This makes a dialog appear which let's you press Ok or Cancel
+		// If you press Ok it will open the SlimeVR installer documentation
+		use rfd::{MessageButtons, MessageDialog, MessageDialogResult, MessageLevel};
+
+		let confirm = MessageDialog::new()
+			.set_title("SlimeVR")
+			.set_description("Couldn't find WebView2 installed. You can install it with the SlimeVR installer")
+			.set_buttons(MessageButtons::OkCancel)
+			.set_level(MessageLevel::Error)
+			.show();
+		if confirm == MessageDialogResult::Ok {
+			open::that("https://docs.slimevr.dev/server-setup/installing-and-connecting.html#install-the-latest-slimevr-installer").unwrap();
+		}
+		return Ok(());
+	}
+	return Ok(());
```

Comment:
again return not needed
```suggestion
	Ok(())
```